onyo/lib/onyo.py

```diff
@@ -337,19 +358,29 @@ def _get_files_untracked(self) -> set[Path]:
         untracked = {Path(x) for x in self._git(['ls-files', '-z', '--others', '--exclude-standard']).split('\0') if x}
         return untracked
 
-    def _get_root(self) -> Path:
+    @staticmethod
+    def _find_root(directory: Union[Path, str]) -> Path:
```

Comment:
Is there a reason, `_find_root` needs to be able to take a `str`?
I do understand, that this is how it's done everywhere, but it's relatively costly to call `Path`'s constructor over and over.